### PR TITLE
Small code tweaks and polish

### PR DESCRIPTION
**Pull request by d4rken**
Improved DownloadTracker tests.
Small codestyle changes and refactoring.

See individual commits for details.

### CODE REVIEW

**Comment by d4rken** [comment]
@dotsbb Would you do a small review before I merge?

File: piwik-sdk/src/main/java/org/piwik/sdk/DownloadTracker.java [hunk 1]
```diff
@@ -105,12 +104,12 @@ public void run() {
 
     private void trackNewAppDownloadInternal(@NonNull Extra extra) {
         Logy.d(LOGGER_TAG, "Tracking app download...");
-        if (mPkgInfo == null) return;
 
         StringBuilder installIdentifier = new StringBuilder();
         installIdentifier.append("http://").append(mPackageName).append(":").append(getVersion());
 
         if (extra == Extra.APK_CHECKSUM) {
+            if (mPkgInfo == null) return;
```
Comment (dotsbb) [inline piwik-sdk/src/main/java/org/piwik/sdk/DownloadTracker.java]: Good catch!

**Comment by dotsbb** [comment]
Nice work! Let's merge when conflicts will be resolved.

**Comment by d4rken** [comment]
Should we squash the commits? Maybe not, as most are independent changes?

**Comment by dotsbb** [comment]
@d4rken squash is not needed in this case baceuse commits are independent and have nice comments

**Pull request merged**